File: solutions/csharp/squeaky-clean/1/SqueakyClean.cs
using System.Text;
using System.Text.Unicode;
public static class Identifier
{
    public static string Clean(string identifier)
    {
        StringBuilder strB = new(identifier);
        strB.Replace(' ', '_');
        strB.Replace("\0", "CTRL");
        for(int i = 1; i < strB.Length; i++)
        {
            if (strB[i - 1] == '-')
                strB[i] = Char.ToUpper(strB[i]);
        }
        

        UnicodeRange greek = UnicodeRanges.GreekandCoptic;
        UnicodeRange greekExtended = UnicodeRanges.GreekExtended;
        bool isGreek(char c)
        {
            return (c >= greek.FirstCodePoint && c <= greek.FirstCodePoint + greek.Length) 
                || (c >= greekExtended.FirstCodePoint && c <= greekExtended.FirstCodePoint + greekExtended.Length);
        }
        for (int i = 0; i < strB.Length; i++)
        {
            Char c = strB[i];
            if (!(Char.IsLetter(c) || c == '_') || (isGreek(c) && Char.IsLower(c)))           
                strB[i] = '-';
        }
        strB.Replace("-", "");
        return strB.ToString();
    }
}

File: solutions/java/squeaky-clean/1/src/main/java/SqueakyClean.java
class SqueakyClean {
    static String clean(String identifier) {
        StringBuilder res = new StringBuilder();
        boolean isNextUpperCase = false;
        for (char c: identifier.toCharArray())
        {
            if (c == ' ')
                res.append('_');
            else if (c == '-')
                isNextUpperCase = true;
            else if (Character.isLetter(c)){
                if (isNextUpperCase) {
                    res.append(Character.toUpperCase(c));
                    isNextUpperCase = false;
                } else
                    res.append(c);
            }
            else if (Character.isDigit(c)) {
                char cNew = switch (c) {
                    case '0' -> 'o';
                    case '1' -> 'l';
                    case '3' -> 'e';
                    case '4' -> 'a';
                    case '7' -> 't';
                    default -> ' ';
                };
                if (cNew != ' ')
                {
                    if (isNextUpperCase) {
                        res.append(Character.toUpperCase(cNew));
                        isNextUpperCase = false;
                    } else
                        res.append(cNew);
                }
            }
        }
        return res.toString();
    }
}
